languages/CPP.groovy

```diff
@@ -0,0 +1,344 @@
+				println(errorMsg)
+				props.error = "true"
+				buildUtils.updateBuildResult(errorMsg:errorMsg,logs:["${member}_bind_pkg.log":bindLogFile])
+			}
+		}
+
+		//perform Db2 Bind Plan
+		bind_performBindPlan = props.getFileProperty('bind_performBindPlan', buildFile)
+		if (bind_performBindPlan && bind_performBindPlan.toBoolean()) {
+			int bindMaxRC = props.getFileProperty('bind_maxRC', buildFile).toInteger()
+			def (bindRc, bindLogFile) = bindUtils.bindPlan(buildFile);
+			if ( bindRc > bindMaxRC) {
+				String errorMsg = "*! The bind plan return code ($bindRc) for $buildFile exceeded the maximum return code allowed ($props.bind_maxRC)"
+				println(errorMsg)
+				props.error = "true"
+				buildUtils.updateBuildResult(errorMsg:errorMsg,logs:["${member}_bind_plan.log":bindLogFile])
+			}
+		}
+	}
+
+    // clean up passed DD statements
+    job.stop()
+}
+
+// end script
+
+
+//********************************************************************
+//* Method definitions
+//********************************************************************
+
+/*
+ * createCParms - Builds up the c compiler parameter list from build and file properties
+ */
+def createCParms(String buildFile, LogicalFile logicalFile) {
+    def parms = props.getFileProperty('cpp_compileParms', buildFile) ?: ""
+    def cics = props.getFileProperty('cpp_compileCICSParms', buildFile) ?: ""
+    def sql = props.getFileProperty('cpp_compileSQLParms', buildFile) ?: ""
+    def compileDebugParms = props.getFileProperty('cpp_compileDebugParms', buildFile)
+
+    if (buildUtils.isCICS(logicalFile))
+        parms = "$parms,$cics"
+
+    if (buildUtils.isSQL(logicalFile))
+        parms = "$parms,$sql"
+
+    // add debug options
+    if (props.debug)  {
+        parms = "$parms,$compileDebugParms"
+    }
+
+    if (parms.startsWith(','))
+        parms = parms.drop(1)
+
+    if (props.verbose) println " C/CPP compiler parms for $buildFile = $parms"
+    return parms
+}
+
+/*
+ * createCompileCommand - creates a MVSExec command for compiling the c program (buildFile)
+ */
+def createCompileCommand(String buildFile, LogicalFile logicalFile, String member, File logFile) {
+    String parms = createCParms(buildFile, logicalFile)
+    String compiler = props.getFileProperty('cpp_compiler', buildFile)
+
+    // define the MVSExec command to compile the program
+    MVSExec compile = new MVSExec().file(buildFile).pgm(compiler).parm(parms)
+
+    // add DD statements to the compile command
+    compile.dd(new DDStatement().name("SYSIN").dsn("${props.cpp_srcPDS}($member)").options('shr').report(true))
+
+    compile.dd(new DDStatement().name("SYSOUT").options(props.cpp_tempOptions))
+    compile.dd(new DDStatement().name("SYSPRINT").options(props.cpp_tempOptions))
+
+    compile.dd(new DDStatement().name("SYSMDECK").options(props.cpp_tempOptions))
+    (1..15).toList().each { num ->
+        compile.dd(new DDStatement().name("SYSUT$num").options("cyl space(1,1) unit(sysallda) new"))
```

Comment:
I don't think we actually need these SYSUTx for the C/CPP compiler. Can you double-check please?

Reply:
Based on this sample [JCL from the IBM docs](https://www.ibm.com/docs/en/adffzscc/1.8.0?topic=programs-sample-jcl-compiling-zos-c), it seems necessary.